crates/bevy_render/src/render_asset.rs

```diff
@@ -61,8 +65,21 @@ pub trait RenderAsset: Send + Sync + 'static + Sized {
     /// ECS data may be accessed via `param`.
     fn prepare_asset(
         source_asset: Self::SourceAsset,
+        asset_id: AssetId<Self::SourceAsset>,
         param: &mut SystemParamItem<Self::Param>,
     ) -> Result<Self, PrepareAssetError<Self::SourceAsset>>;
+
+    /// Called whenever the [`RenderAsset::SourceAsset`] has been removed.
+    ///
+    /// You can implement this method if you need to access ECS data (via
+    /// `_param`) in order to perform cleanup tasks when the asset is removed.
+    ///
+    /// The default implementation does nothing.
+    fn finalize_asset(
+        _source_asset: AssetId<Self::SourceAsset>,
+        _param: &mut SystemParamItem<Self::Param>,
+    ) {
+    }
```

Comment:
To me `finalize_asset` could also sound like the last step that happens before the asset is ready to use. The comment was very helpful though.

Reply:
Yes I'd rather rename it on_unload or something.

Reply:
It's now called `unload_asset` to go along with `prepare_asset`.